### PR TITLE
Feature/e2e

### LINKED ISSUE #10: Add end to end -tests
#### Proposed solution

Create tests for testing UI, frontend logic and backend together. This can be done with either Selenium or Cypress (I'd prefer latter, it's easier to use) and docker (for running backend and db).

End to end tests should also be integrated to CI pipeline. 

#### DoD (Definition of Done)

Core functionalities are tested.

### PR DESCRIPTION
**Pull request by saulipurhonen**
### Description

Basic e2e test that covers:
- Folder creation
- Form save
- XML upload
- Wizard navigation
- Folder publishing

To run, execute **npx cypress open**

### Related issues

Closes #10 

### Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update (this needs a follow up PR)

### Changes Made

- Installed Cypress and cypress-file-upload packages
- Tested frontend logic with backend connection

### Testing

- [x] Tests do not apply

### CODE REVIEW

**Comment by saulipurhonen** [comment]
Note: There isn't much more to test in current state of the app. This merge can be used as base for e2e tests with Cypress.

Also, CI and GH Actions integration aren't in this PR. Should we try to get those running as well?

**Comment by blankdots** [comment]
> Also, CI and GH Actions integration aren't in this PR. Should we try to get those running as well?

yes I think those should be done as well this could be introduced as part of this PR or another PR  (if that is the case please create an issue about what will be done)
https://github.com/cypress-io/github-action - seems to be the way to go ?

Note: the workflow will change following https://github.com/CSCfi/metadata-submitter/pull/158 and https://github.com/CSCfi/metadata-submitter/pull/151

as the user would need login, but a mock AAI is provided in the integration tests of those PRs.

File: cypress/integration/app.spec.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    cy.get("button[type=button]").contains("Next").click()
+
+    // Fill a study form and save object
+    cy.get("div[role=button]").contains("Study").click()
+    cy.get("div[role=button]").contains("Fill Form").click()
+    cy.get("input[name='descriptor.studyTitle']").type("Test title")
+    cy.get("select[name='descriptor.studyType']").select("Metagenomics")
+    cy.get("button[type=submit]").contains("Save").click()
+
+    // Upload an xml file. Need to wait for save. TODO: Better logic for waiting until last save is done
```
Comment (blankdots) [inline cypress/integration/app.spec.js:25]: please do an issue detailing this To Do.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -6,6 +6,8 @@ If you're reading this, it means you saw something that is not right, you want t
 
 We have a handful of unit tests and integration tests. In our Git workflow unit tests are run on every Push and Pull Request, while integration tests run on every Pull Request.
 
+We use Cypress for e2e testing.
```
Comment (blankdots) [inline CONTRIBUTING.md:9]:
```suggestion
We use [Cypress](https://www.cypress.io/) for e2e testing.
```

**Review by blankdots — CHANGES_REQUESTED** [review]
I am ok with leaving `This example ...` in the cypress files but I must recognise don't look very nice.

We would reference the source and where to find more information.

the only requests of change would be:
- to have a github action for this.
- mention in main `README.md` how to run the tests

**Comment by saulipurhonen** [comment]
> I am ok with leaving `This example ...` in the cypress files but I must recognise don't look very nice.
> 
> We would reference the source and where to find more information.
> 
> the only requests of change would be:
> 
> * to have a github action for this.
> * mention in main `README.md` how to run the tests

I agree that example file doesn't look good.
I Removed the file and added e2e testing info to README.md

**Review by csc-jm — APPROVED** [review]
LGTM

**Review by blankdots — APPROVED** [review]

**Pull request merged**